File: src/main/java/com/epri/dss/shared/impl/StackBaseImpl.java
package com.epri.dss.shared.impl;

import com.epri.dss.shared.StackBase;

public class StackBaseImpl implements StackBase {

	protected int numItems, increment, maxItems;

	public StackBaseImpl(int initSize) {
		super();
		this.maxItems = initSize;
		this.increment = initSize;
		this.numItems = 0;
	}

	public void clear() {
		numItems = 0;
	}

	public int size() {
		return numItems;
	}

}

File: pydss/shared/StackBaseImpl.py
from dss.shared.StackBase import (StackBase,)


class StackBaseImpl(object, StackBase):
    numItems = None
    increment = None
    maxItems = None

    def __init__(self, initSize):
        super(StackBaseImpl, self)()
        self.maxItems = initSize
        self.increment = initSize
        self.numItems = 0

    def clear(self):
        self.numItems = 0

    def size(self):
        return self.numItems
